### DISCUSSION TITLE
invalid command in gajoe while using external pool

### DISCUSSION THREAD

**Discussion by zzzichenkeke** [biosaxs-com/atsas-community#1941] [Mixtures and flexible systems] [answered]
Hi, I met an issue that, when I tried to use command gajoe -p/gajoe --pool to fit my external pool in, it reported an error like: gajoe -p
*******  ------------------------------------------------------  *******
*******     GAJOE - version 2.1 - (r14885)                       *******
*******     Copyright (c) ATSAS Team                             *******
*******     EMBL, Hamburg Outstation, 2007 - 2015                *******
*******                                                          *******
*******     For doubts/questions please visit SAXIER forum:      *******
*******     http://www.saxier.org/forum/viewforum.php?f=10       *******
*******                                                          *******
*******     In case of bugs please refer to:                     *******
*******     G. Tria, D.I. Svergun, EMBL BioSAXS group            *******
*******     atsas@embl-hamburg.de                                *******
*******  ------------------------------------------------------  *******
 getopt error: unknown option: -p

Is there any clue what could cause this error? Did i misuse the option? I really appreciate your help!

**Comment by franke-biosaxs** [comment, marked as answer]
The interface of gajoe changed in 3.2.x, IIRC. If you can, please update to ATSAS-4.0.0.
Further, all applications have the -h/--help options to list their options.

Besides that, I think you probably want:
 1. obtain your pool
 2. `ffmaker --pool=size.csv --output=intensities.csv [models]`
 3. `gajoe --intensity=intensity.csv --size=size.csv [data]`

The above is from memory and untested right now. But it should work this way or similar.

**Reply by zzzichenkeke** [reply]
Hi, thank you so much for your kind reply. Sorry I couldn't find the latest ATSAS-4.0.0 from this webpage: https://www.embl-hamburg.de/biosaxs/atsas-online/download.php. Am I in the wrong one? And the commands you offered seem work well with my models, thanks again!

**Reply by franke-biosaxs** [reply]
[ATSAS 4.0 is right here, on GitHub.](https://github.com/biosaxs-com/atsas-community/releases)

**Reply by zzzichenkeke** [reply]
Thank you so much!
